### ISSUE TITLE
Pi-hole not responding to clients asking to Pi-hole hostname

### ISSUE THREAD

**Issue by Mic-Pod** [pi-hole/FTL#1191]
Hi,
Before recent updates clients were able to resolve Pi-hole server's hostname with IP. Currently client can't resolve Pi-hole server's hostname.
In web GUI, there are no entries in Query Log when doing something like `dig @192.168.1.100 server1` but there are entries in /var/log/pihole.log (details below).

### Before:
`dig +short @192.168.1.100 server1`
`192.168.1.100`

### After:
`dig +short @192.168.1.100 server1`

### Versions

- Pi-hole:  5.5
- AdminLTE:  5.7
- FTL:  5.10.2

### Platform

- OS and version:  Ubuntu Server 18.04.6 LTS

- Platform:  

### Expected behavior

Resolving pi-hole hostname with IP

### Actual behavior / bug

No answer with IP when asking for pi-hole hostname

### Steps to reproduce

`dig @192.168.1.100 server1`
no ANSWER SECTION
entry in /var/log/pihole.log:
`Oct  1 22:45:27 dnsmasq[23259]: query[A] server1 from 192.168.1.11
Oct  1 22:45:27 dnsmasq[23259]: Pi-hole hostname server1 is Pi-hole hostname`

**Comment by DL6ER** [comment]
Thanks for your report and sorry for the inconvenience caused by this. You should be able to workaround this by manually setting a suitable `REPLY_ADDR4` and/or `REPLY_ADDR6` in `/etc/pihole/pihole-FTL.conf`. See also the discussion in our Discourse forum posted by @pralor-bot directly above my comment.

Could you please add
```
DEBUG_QUERIES=true
DEBUG_NETWORKING=true
```
to `/etc/pihole/pihole-FTL.conf` and send me the snippet from `/var/log/pihole-FTL.log` when doing the `dig pi.hole` ?
There should be a line like
```
Interfaces: Called from ...
```
and one like
```
**** new UDP IPv4 query[A] query "pi.hole" from ...
```

Ideally, there is content in between. If there is noting in between these lines, Pi-hole isn't able to read the interfaces on your system. On which kind of hardware are your system running? Do you use any kind of virtualization?

**Comment by DL6ER** [comment]
@Mic-Pod @AlphaCluster Please also run

pihole checkout ftl tweak/iface_detection

and repeat the `dig pi.hole` and send me what you see in `pihole-FTL.log`.

I added more debugging output and this should hopefully reveal why your interface is not correctly being detected here.

**Comment by Mic-Pod** [comment]
I have Pi-hole installed on x86 baremetal (NUC).
When doing dig for Pi-hole hostname there are entries in pihole-FTL.log - `Replying to mic-nuc with NODATA due to missing iface address`

About REPLY_ADDR4 option - what value should I set it to?

**Comment by bcutter** [comment]
Same issue on a Raspberry Pi 4 running on Raspbian (32 bit).

**Comment by DL6ER** [comment]
@Mic-Pod Did you try after running the
```
pihole checkout ftl tweak/iface_detection
```
?

If not, could you try this?

@bcutter and anyone else who is affected, too. The more input we can get here, the better.

**Comment by Mic-Pod** [comment]
@DL6ER Yes - I run the `pihole checkout ftl tweak/iface_detection` and set
```
DEBUG_QUERIES=true
DEBUG_NETWORKING=true
```
in `pihole-FTL.conf`.

**Comment by DL6ER** [comment]
You should see stuff like
```
[2021-10-02 13:59:26.683 143101M] Interfaces: Called from src/dnsmasq/forward.c:1549
[2021-10-02 13:59:26.683 143101M] Identifying interface (looking for 127.0.0.1):
[2021-10-02 13:59:26.683 143101M]   - IPv4 interface veth7fd0c2e (32,0) is 169.254.109.194
[...]
[2021-10-02 13:59:26.684 143101M] Analyzing interfaces:
[2021-10-02 13:59:26.684 143101M]   - SKIP IPv4 interface veth7fd0c2e: (32,0) != (1,0)
[...]
```
do you see these? Can you quote them? If them contain public addresses you don't want to show here, you could copy-paste this into a dedicated text file and run
```
cat /tmp/textfile | pihole tricorder
```
to get this uploaded to our support server. It will get auto-deleted after 48 hours and I'd only need the token.

**Comment by Mic-Pod** [comment]
Sorry, there are no such entries in pihole-FTL.log and pihole.log files.

**Comment by Mic-Pod** [comment]
https://tricorder.pi-hole.net/sncr2IoF/

**Comment by DL6ER** [comment]
@Mic-Pod Okay, when there aren't any such lines, this explains why the interface is not found - it is not analyzed. Could you update the branch and add
```
DEBUG_FLAGS=true
```
to `/etc/pihole/pihole-FTL.conf` and restart `pihole-FTL` once again? 

You should now see two lines like
```
[2021-10-02 14:06:53.703 143101M] Processing FTL hook from src/dnsmasq/forward.c:1574...
[2021-10-02 14:06:53.703 143101M]      Flags: F_FORWARD F_IPV4 F_QUERY 
```
before the line
```
[2021-10-02 14:06:53.703 143101M] Replying to mic-nuc with NODATA due to missing iface address
```

It'll be interesting to see if you see the same lines and if the lines will also indicate `forward.c:1574` as source.

**Comment by Mic-Pod** [comment]
Wait a second... Suddenly it started to work.... I don't get it...
I run `pihole checkout ftl tweak/iface_detection` before adding `DEBUG_FLAGS=true` but noticed after that that there is a correct response in dig now.

**Comment by DL6ER** [comment]
> I don't get it...

I generalized the interface analyzing routine and made it cover some edge-cases (where I thought none would apply to you). If that fixed the issue all the better!

Thanks also for the Tricorder token. Now we know what is going on, you have a file that enables `bind-interfaces`:
```
lrwxrwxrwx 1 root root 28 Oct  1  2018 /etc/dnsmasq.d/lxd -> /etc/dnsmasq.d-available/lxd
   bind-interfaces
   except-interface=lxdbr0
```

You may want to delete this file if you didn't install it yourself. This option is known to cause all sorts of issues and the official `dnsmasq` documentation says:
> About the only time when this is useful is when running another nameserver (or another instance of dnsmasq) on the same machine.

**Comment by Mic-Pod** [comment]
OK, I have removed lxd config file (don't need it) returned to master and it's working fine.
Thank You for Your time during this troubleshooting ;)

**Comment by DL6ER** [comment]
> However there are still no entries in Query Log (web-gui)

There was never an entry showing `pi.hole`. We hide lookups to the hostname as well now as they're taking a shortcut and do not go through the embedded `dnsmasq`. Do you need this or is it just something you have noticed?

**Comment by Mic-Pod** [comment]
> > However there are still no entries in Query Log (web-gui)
> 
> There was never an entry showing `pi.hole`. We hide lookups to the hostname as well now as they're taking a shortcut and do not go through the embedded `dnsmasq`. Do you need this or is it just something you have noticed?

OK, I didn't know that. It's not something I need. I just wanted to know if this is a correct behavior.

**Comment by yubiuser** [comment]
> OK, I didn't know that.

We're going to change that https://github.com/pi-hole/AdminLTE/pull/1913 ;-)

**Comment by jfb-pihole** [comment]
"Refusing to answer for certain hosts"

In the specific cases of the Firefox canary domain and the Apple Private Relay domains, we have implemented blocking exactly as prescribed by the developers of those specific OS'S and packages.
